src/mlpack/methods/random_forest/random_forest_impl.hpp

```diff
@@ -507,6 +507,21 @@ double RandomForest<
   #pragma omp parallel for reduction( + : totalGain)
   for (size_t i = 0; i < numTrees; ++i)
   {
+    // NOTE: this is a hacky workaround for older versions of Armadillo that did
+    // not (by default) set a different seed for each RNG.  We simply manually
+    // set the RNG seed for each individual thread.  However, if users have
+    // specifically set the RNG seed for Armadillo, this could break their code.
+    // So we hide it behind an (undocumented except for here) ifndef so that
+    // this support can be disabled if desired...
+    #if (ARMA_VERSION_MAJOR < 12) || \
+        ((ARMA_VERSION_MAJOR == 12) && (ARMA_VERSION_MINOR < 6))
```

Comment:
This may also have to take into account ARMA_VERSION_PATCH.  The fix will be in Armadillo 12.6.2, so 12.6.1 and 12.6.0 are still "bad" versions.

Reply:
Actually I don't want to apply the workaround in all cases.  The workaround isn't to set specific threads to a specific seed, but instead just to set the seeds to something that will be different for each thread (thus working around the fact that old Armadillo will have the same seeds set).  Two reasons I'd like to leave it as a workaround:

1. So that it's obvious when the code can be removed (as soon as the minimum Armadillo version is >= 12.6.2).

2. Because the Armadillo RNG is being set with mlpack's RNG, users could possibly encounter situations where they set the Armadillo RNG specifically (but not the mlpack RNG), and then the mlpack RNG overwrites the Armadillo RNG seeds.  This would result in a situation where the user setting the Armadillo RNG actually does not make a difference in their code!  It's a little bit of a contrived situation, but I'd prefer for it to not be possible.

Anyway, it's a minor point, and reason (1) is the stronger reason here.